### PR TITLE
EUREKA-1154 Return 404 if instance is not found when updating metadata

### PR DESCRIPTION
**Pull request by mgtriffid**

### CODE REVIEW

**Comment by mgtriffid** [comment]
Fixes #1154.

File: eureka-core/src/main/java/com/netflix/eureka/resources/InstanceResource.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public Response renewLease(
         }
         // Check if we need to sync based on dirty time stamp, the client
         // instance might have changed some value
-        Response response = null;
```
Comment (PropertyOfTuckerSystems) [inline eureka-core/src/main/java/com/netflix/eureka/resources/InstanceResource.java]: This change is unrelated and I'd suggest to submit a separate PR for style only fixes if you're interested.

File: eureka-core/src/main/java/com/netflix/eureka/resources/InstanceResource.java [hunk 2]
```diff
@@ -239,15 +239,15 @@ public Response updateMetadata(@Context UriInfo uriInfo) {
...
-                return Response.serverError().build();
+                logger.warn("Cannot find instance while updating metadata for instance {}/{}", app.getName(), id);
+                return Response.status(Status.NOT_FOUND).build();
             }
             MultivaluedMap<String, String> queryParams = uriInfo.getQueryParameters();
             Set<Entry<String, List<String>>> entrySet = queryParams.entrySet();
             Map<String, String> metadataMap = instanceInfo.getMetadata();
             // Metadata map is empty - create a new map
             if (Collections.emptyMap().getClass().equals(metadataMap.getClass())) {
-                metadataMap = new ConcurrentHashMap<String, String>();
```
Comment (PropertyOfTuckerSystems) [inline eureka-core/src/main/java/com/netflix/eureka/resources/InstanceResource.java]: Again would suggest to separate, but I understand these are minor so up to you.
